front/src/App.vue

```diff
@@ -0,0 +1,247 @@
+<template>
+    <div id="app">
+        <login v-on:login="onLogin($event)" v-if="!isConnected"></login>
+        <room-editor v-if="!roomEditorHidden" v-model:hidden="roomEditorHidden" v-on:create-room="createRoom($event)" v-on:join-room="joinRoom($event)"></room-editor>
+        <div class="app_vertical_layout">
+            <div class="app_top_horizontal_layout">
+                <room-list v-model:rooms="rooms" v-on:room-selected="displayRoom($event)" v-on:room-quit="quitRoom($event)"></room-list>
+                <div class="app_messages_list_wrapper">
+                    <activity-bar v-bind:current-room="selectedRoom" v-bind:is-writing="isWriting"></activity-bar>
+                    <message-list v-bind:messages="messages"></message-list>
+                </div>
+                <user-list v-bind:users="users"></user-list>
+            </div>
+            <div class="app_bottom_horizontal_layout">
+                <div class="app_room_editor_start" @click="roomEditorHidden = false">
+                    <div class="app_room_editor_plus"></div>
+                </div>
+                <message-editor v-model:content="currentTypedMessage" v-on:send-message="onMessageSend($event)" v-on:start-writing="isWriting = true" v-on:stop-writing="isWriting = false"></message-editor>
+            </div>
+        </div>
+    </div>
+</template>
```

Comment:
make a commentarie by parts like for this`<!-- app template  -->`

Reply:
I mean its the `<template>` part, of the App.vue file, this doesn't need comment you just need to know vue.

Reply:
No comment needed here, it's crystal clear.